integration_tests/commands/async/jsonresp_test.go

```diff
@@ -6,6 +6,36 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func arraysArePermutations[T comparable](a, b []T) bool {
+	// If lengths are different, they cannot be permutations
+	if len(a) != len(b) {
+		return false
+	}
+
+	// Count occurrences of each element in array 'a'
+	countA := make(map[T]int)
+	for _, elem := range a {
+		countA[elem]++
+	}
+
+	// Subtract occurrences based on array 'b'
+	for _, elem := range b {
+		countA[elem]--
+		if countA[elem] < 0 {
+			return false
+		}
+	}
+
+	// Check if all counts are zero
+	for _, count := range countA {
+		if count != 0 {
+			return false
+		}
+	}
+
+	return true
+}
+
```

Comment:
I would like to confirm that this change in async is out of the scope of the PR but is needed to keep existing functionality intact.

Reply:
Yes, this is required for JSON.RESP tests. I see this command is in progress hence have kept it there. We can clean it up once all JSON commands are migrated.